packages/sdk/src/internal/Extensions/IntegrationAdapters/ArrakisV2.ts

```diff
@@ -0,0 +1,82 @@
+import { type Address, type Hex, decodeAbiParameters, encodeAbiParameters } from "viem";
+import * as IntegrationManager from "../../IntegrationManager.js";
+
+//--------------------------------------------------------------------------------------------
+// LEND
+//--------------------------------------------------------------------------------------------
+
+const lendSelector = "0x099f7515"; // lend(address,bytes,bytes)
+export const lend = IntegrationManager.makeUse(lendSelector, lendEncode);
+
+const lendEncoding = [
+  {
+    type: "address",
+    name: "arrakisVault",
+  },
+  {
+    name: "maxUnderlyingAmounts",
+    type: "uint256[2]",
+  },
+  {
+    name: "sharesAmount",
+    type: "uint256",
+  },
+] as const;
+
+export type LendArgs = {
+  arrakisVault: Address;
+  maxUnderlyingAmounts: [bigint, bigint];
+  sharesAmount: bigint;
+};
+
+export function lendEncode(args: LendArgs): Hex {
+  return encodeAbiParameters(lendEncoding, [args.arrakisVault, args.maxUnderlyingAmounts, args.sharesAmount]);
+}
+
+export function lendDecode(encoded: Hex): LendArgs {
+  const [arrakisVault, maxUnderlyingAmounts, sharesAmount] = decodeAbiParameters(lendEncoding, encoded);
+
+  return {
+    arrakisVault,
+    maxUnderlyingAmounts: [...maxUnderlyingAmounts],
```

Comment:
Why?

Reply:
to get rid off the read only modifier

Reply:
In other parts of the SDK, we type the return value as `Readonly`, e.g. here: https://github.com/enzymefinance/sdk/blob/main/packages/sdk/src/internal/Extensions/IntegrationAdapters/BalancerV2Liquidity.ts#L59

We should try to be consistent.